### PR TITLE
Integration with mosaic.js npm - 0.10.0-beta.2

### LINKED ISSUE #76: Integrate mosaic.js npm module instead of mosaic-tbd npm module in package.json

### PR DESCRIPTION
**Pull request by abhayks1**
Integration of mosaic.js npm **0.10.0-beta.2**
Fixes #76

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "pre-commit": "lint-staged"
   },
   "dependencies": {
-    "@openstfoundation/mosaic-tbd": "https://github.com/OpenSTFoundation/mosaic-tbd.git"
+    "@openstfoundation/mosaic.js": "0.10.0-beta.2"
```
Comment (benjaminbollen) [inline package.json]: is this maybe a `peer-dependency`?

Comment (benjaminbollen) [reply package.json]: it is not.

File: lib/AbiBinProvider.js [hunk 2]
```diff
@@ -5,8 +5,8 @@ const fs = require('fs'),
   path = require('path');
 //__NOT_FOR_WEB__END__
 
-const Mosaic = require('@openstfoundation/mosaic-tbd');
-const AbiBinProvider = Mosaic.AbiBinProvider;
+const Mosaic = require('@openstfoundation/mosaic.js');
+const { AbiBinProvider } = Mosaic;
```
Comment (schemar) [inline lib/AbiBinProvider.js]: Do you need `Mosaic` further down? If not you could merge the above two lines into:
```suggestion
const { AbiBinProvider } = require('@openstfoundation/mosaic.js');
```

File: lib/AbiBinProvider.js [hunk 3]
```diff
@@ -24,16 +24,30 @@ class OpenSTAbiBinProvider extends AbiBinProvider {
    *
    * @param abiFolderPath Folder path where abi is present.
    * @param binFolderPath Folder path where bin is present.
```
Comment (schemar) [inline lib/AbiBinProvider.js]: Should be marked optional, right? `{string} [abiFolderPath]` and `{string} [binFolderPath]`.

File: lib/Contracts.js [hunk 4]
```diff
@@ -2,8 +2,8 @@
 
 const Web3 = require('web3');
 const AbiBinProvider = require('./AbiBinProvider');
-const Mosaic = require('@openstfoundation/mosaic-tbd');
-const Contracts = Mosaic.Contracts;
+const Mosaic = require('@openstfoundation/mosaic.js');
+const { Contracts } = Mosaic;
```
Comment (schemar) [inline lib/Contracts.js]: Same as `AbiBinProvider`

File: lib/AbiBinProvider.js (hunk 2 above)
Comment (abhayks1) [reply lib/AbiBinProvider.js]: Done!

File: lib/AbiBinProvider.js (hunk 3 above)
Comment (abhayks1) [reply lib/AbiBinProvider.js]: Done!

File: lib/Contracts.js (hunk 4 above)
Comment (abhayks1) [reply lib/Contracts.js]: Done!

File: lib/AbiBinProvider.js [hunk 5]
```diff
@@ -22,18 +21,32 @@ class OpenSTAbiBinProvider extends AbiBinProvider {
   /**
```
Comment (pgev) [inline lib/AbiBinProvider.js]: I am not sure why OpenSTAbiBinProvider is inheriting from AbiBinProvider of mosaic?

File: lib/AbiBinProvider.js [hunk 6]
```diff
@@ -22,18 +21,32 @@ class OpenSTAbiBinProvider extends AbiBinProvider {
...
-   * @param mosaicBinFolderPath Folder path where mosaic bin is present.
+   * @param {string} [abiFolderPath] Folder path where abi is present.
+   * @param {string} [binFolderPath] Folder path where bin is present.
    */
-  constructor(abiFolderPath, binFolderPath, mosaicAbiFolderPath, mosaicBinFolderPath) {
-    abiFolderPath = abiFolderPath || DEFAULT_ABI_FOLDER_PATH;
-    binFolderPath = binFolderPath || DEFAULT_BIN_FOLDER_PATH;
-    super(abiFolderPath, binFolderPath);
+  constructor(abiFolderPath, binFolderPath) {
+    const abiDirectoryPath = abiFolderPath || DEFAULT_ABI_FOLDER_PATH;
```
Comment (pgev) [inline lib/AbiBinProvider.js]: DEFAULT_ABI_FOLDER_PATH and DEFAULT_BIN_FOLDER_PATH are guarded (__NOT_FOR_WEB__BEGIN__). Should we check that if they are undefined and abiFolderPath, binFolderPath are not passed and raise error?

File: lib/AbiBinProvider.js (hunk 5 above)
Comment (abhayks1) [reply lib/AbiBinProvider.js]: OpenST is on top of mosaic. Inheriting  AbiBinProvider of mosaic helps end users to retrieve abi/bin of mosaic with a single OpenST AbiBinProvider object.

File: lib/AbiBinProvider.js (hunk 6 above)
Comment (abhayks1) [reply lib/AbiBinProvider.js]: Makes sense. However I am still not sure how can DEFAULT_ABI_FOLDER_PATH, DEFAULT_BIN_FOLDER_PATH can be undefined. 

Having said that once openst-contracts npm package are integrated, we will not need path dependencies .

File: lib/AbiBinProvider.js (hunk 5 above)
Comment (schemar) [reply lib/AbiBinProvider.js]: But should they need to? Why?

File: lib/AbiBinProvider.js (hunk 6 above)
Comment (schemar) [reply lib/AbiBinProvider.js]: @abhayks1 can you please expand on that? I cannot follow.

**Pull request merged**